### PR TITLE
GitHub issue 138

### LINKED ISSUE #138: In a Multi-Configuration project, if a MATLAB version does not exist in the system, no error or warning generated
For example, if I dont have version R2018a in the system, in a Free-style project, I get a warning message in the UI Configuration page "Unable to find MATLAB on this machine using the specified location. The specified root folder might contain MATLAB on a different build agent"

In a Multi-config project, no such error or warning at the end of the build.

### PR DESCRIPTION
**Pull request by nbhoski**
This PR closes #138 
* UseMatlabVersion now verifies if MATLAB executable is present at a given location.
* If MATLAB executable could not be found then throws Error on Jenkins console and fails the build.
* Error is thrown even if Already some MATLAB is in the Path and user provides MATLAB version explicitly using UseMatlabVersion
* If MATLAB is not explicitly provided using UseMatlabVersion then default MATLAB on the Path will be picked.
* Added dummy executable files in test resources.
* Updated the failed test and added new to verify the functionality.

### CODE REVIEW

File: src/main/resources/config.properties [hunk 1]
```diff
@@ -12,6 +12,7 @@ Builder.matlab.test.support.error = To run tests with the Jenkins plugin, use MA
 builder.matlab.automatictestoption.display.name = Automatic
 builder.matlab.customcommandoption.display.name = Custom
 Releaseinfo.matlab.version.not.found.error = Error finding MATLAB release for given MATLAB root. Verify MATLAB root path.
+matlab.not.found.error = Error finding MATLAB at given location. Verify MATLAB root path.
```
Comment (mw-hrastega) [inline src/main/resources/config.properties:15]: Can we use this text?

Unable to launch MATLAB from the specified location. Verify the path to MATLAB root folder.

Comment (nbhoski) [reply src/main/resources/config.properties:15]: Done.

**Review by mw-hrastega — APPROVED** [review]

File: src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java [hunk 2]
```diff
@@ -133,18 +134,20 @@ public synchronized void setUp(Context context, Run<?, ?> build, FilePath worksp
...
+                getLocalMatlab() + "/bin/" + getNodeSpecificExecutable(launcher));
+
+        if (!matlabExecutablePath.exists()) {
+            throw new MatlabNotFoundError(Message.getValue("matlab.not.found.error"));
+        }
         // Add "matlabroot" without bin as env variable which will be available across the build.
         context.env("matlabroot", getLocalMatlab());
         // Add matlab bin to path to invoke MATLAB directly on command line.
-        context.env("PATH+matlabroot", getLocalMatlab() + nodeSpecificFileSep + "bin");     
+        context.env("PATH+matlabroot", getLocalMatlab() + "/bin");
```
Comment (mw-akumar) [inline src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java:147]: Would it be better to just matlabExecutablePath.parent() ? This way we are using the FilePath abstraction and reusing our previously computed matlabExecutablePath?

File: src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java [hunk 3]
```diff
@@ -133,18 +134,20 @@ public synchronized void setUp(Context context, Run<?, ?> build, FilePath worksp
         // Set Environment variable
 
         setEnv(initialEnvironment);
-        String nodeSpecificFileSep = getNodeSpecificFileSeperator(launcher);
```
Comment (mcafaro) [inline src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java:136]: Why don't we need to be concerned with the node specific file separator now?

File: src/main/resources/config.properties [hunk 4]
```diff
@@ -12,6 +12,7 @@ Builder.matlab.test.support.error = To run tests with the Jenkins plugin, use MA
 builder.matlab.automatictestoption.display.name = Automatic
 builder.matlab.customcommandoption.display.name = Custom
 Releaseinfo.matlab.version.not.found.error = Error finding MATLAB release for given MATLAB root. Verify MATLAB root path.
+matlab.not.found.error = Unable to launch MATLAB from the specified location. Verify the path to MATLAB root folder.
```
Comment (mcafaro) [inline src/main/resources/config.properties:15]: @mw-hrastega should take a look at this.

**Review by mcafaro — APPROVED** [review]
Overall looks good. Just a question about why we can always use a "/" file sep now and couldn't before. Houman should probably take a look at the error message as well.

File: src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java (hunk 3 above)
Comment (nbhoski) [reply src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java:136]: I learnt it late that FilePath handles "/" nd after initial draft did not get chance to touch this file :)

File: src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java (hunk 2 above)
Comment (nbhoski) [reply src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java]: Done.

**Pull request merged**